api/v1alpha1/inferencepool_types.go

```diff
@@ -61,8 +61,100 @@ type InferencePoolSpec struct {
 	// +kubebuilder:validation:Maximum=65535
 	// +kubebuilder:validation:Required
 	TargetPortNumber int32 `json:"targetPortNumber"`
+
+	// EndpointPickerConfig selects and configures the endpoint picking algorithm to apply on the requests sent
+	// to this pool.
+	EndpointPickerConfig `json:"endpointPickerConfig"`
+}
+
+// EndpointPickerConfig selects and configures the endpoint picking algorithm to apply on the requests sent.
+// Only one of the following options can be set.
+type EndpointPickerConfig struct {
+	// Extension configures an endpoint picker as an extension service.
+	//
+	// +optional
+	Extension *ExtensionConfig `json:"extension"`
+	// Algorithm configures the endpoint picker by a name that the provider understands
+	// and knows how to set it up with its own gateway implementation.
+	//
+	// +optional
+	Algorithm *AlgorithmConfig `json:"algorithm"`
+}
+
+// AlgorithmConfig is a reference to the endpoint picking algorithm that the gateway should apply.
+type AlgorithmConfig struct {
+	// Name is an identifier of the algorithm that the gateway should employ.
+	Name string `json:"name"`
+}
+
+// ExtensionConfig specifies how to configure an extension that implements the endpoint picking
+// algorithm.
+type ExtensionConfig struct {
+	// ExtensionRef is a reference to a deployed extension.
+	ExtensionRef *ExtensionReference `json:"extensionRef"`
+
+	// ExtensionConnection configures the connection between the gateway and the extension.
+	ExtensionConnection `json:"extensionConnection"`
+}
+
+// ExtensionReference is a reference to the extension deployment.
+type ExtensionReference struct {
+	// Group is the group of the referent.
+	// When unspecified or empty string, core API group is inferred.
+	//
+	// +optional
+	// +kubebuilder:default=""
+	Group *string `json:"group,omitempty"`
+
+	// Kind is the Kubernetes resource kind of the referent. For example
+	// "Service".
+	//
+	// Defaults to "Service" when not specified.
+	//
+	// ExternalName services can refer to CNAME DNS records that may live
+	// outside of the cluster and as such are difficult to reason about in
+	// terms of conformance. They also may not be safe to forward to (see
+	// CVE-2021-25740 for more information). Implementations SHOULD NOT
+	// support ExternalName Services.
+	//
+	// Support: Core (Services with a type other than ExternalName)
+	//
+	// Support: Implementation-specific (Services with type ExternalName)
+	//
+	// +optional
+	// +kubebuilder:default=Service
+	Kind *string `json:"kind,omitempty"`
+
+	// Name is the name of the referent.
+	Name string `json:"name"`
+
+	// The port number on the pods running the extension. Defaults to 9002 if not set.
+	//
+	// +kubebuilder:default=9002
+	TargetPortNumber *int32 `json:"targetPortNumber"`
```

Comment:
xRef types should only include group, kind, name, and optionally namespace. Suppose targetPortNumber is to be part of this config API. In that case, we should state that the ExtensionReference should not include targetPortNumber or; if it does, it would override the targetPortNumber defined in ExtensionConfig.

Reply:
> then we can't spec how it is setup in the InferencePool API, right?

Then we would need to support [ReferenceGrant](https://gateway-api.sigs.k8s.io/api-types/referencegrant/).

Reply:
ok, so we agree that this is out of scope for this api, right?

Reply:
I think this thread is really covering two concepts:

1. Does the target of this reference want to accept these connections?
2. Can we secure these connections?

On 1, we've historically considered resources within the same namespace as trusted by default, and introduced RefGrant to allow for safe cross-namespace references. I'd agree that that should be out of scope until we have a more concrete need for it.

On 2, I could see a rationale for some form of (m)TLS here, but that seems like it would require non-trivial extra work. I'd rather track that as a follow up item to explore.

Reply:
Actually apparently we're already working on TLS config for EPP (https://github.com/kubernetes-sigs/gateway-api-inference-extension/issues/225), so then the question becomes if anyone wants mTLS here?

Reply:
> On 2, I could see a rationale for some form of (m)TLS here, but that seems like it would require non-trivial extra work. I'd rather track that as a follow up item to explore.

+1 @ahg-g do you mind creating a tracking issue so we don't forget about this?

Reply:
https://github.com/kubernetes-sigs/gateway-api-inference-extension/issues/260